tests/unit/package_managers/test_generic.py

```diff
@@ -39,10 +75,79 @@ def test_fetch_generic_source(
     mock_from_obj_list.assert_called_with(components=components)
 
 
-def test_resolve_generic_no_lockfile(rooted_tmp_path: RootedPath) -> None:
+@mock.patch("cachi2.core.package_managers.generic.main._load_lockfile")
+def test_resolve_generic_no_lockfile(mock_load: mock.Mock, rooted_tmp_path: RootedPath) -> None:
     with pytest.raises(PackageRejected) as exc_info:
         _resolve_generic_lockfile(rooted_tmp_path, rooted_tmp_path)
     assert (
         f"Cachi2 generic lockfile '{DEFAULT_LOCKFILE_NAME}' missing, refusing to continue"
         in str(exc_info.value)
     )
+    mock_load.assert_not_called()
+
+
+@pytest.mark.parametrize(
+    ["lockfile", "expected_exception", "expected_err"],
+    [
+        pytest.param("{", PackageRejected, "yaml format is not correct", id="invalid_yaml"),
+        pytest.param(
+            LOCKFILE_WRONG_VERSION, PackageManagerError, "Input should be '1.0'", id="wrong_version"
+        ),
+        pytest.param(
+            LOCKFILE_NO_CHECKSUM_1, PackageManagerError, "Field required", id="no_checksum_1"
```

Comment:
"no_checksum_1" does not really describe the test case. Would "checksum_field_is_missing" be a better desctiption?